modules/ravestate/context.py

```diff
@@ -523,9 +564,25 @@ def _complete_signal(self, sig: Signal, known_signals: Set[Signal]) -> Optional[
         known_signals.discard(sig)
         return result if len(result) else None
 
+    def _update_core_properties(self):
+        with self._lock:
+            activation_pressure_present = any(activation.is_pressured() for activation in self._state_activations())
+            # don't count states that have ':activity:changed' in their constraint to avoid self-influencing
+            number_of_partially_fulfilled_states = \
+                sum(1 if any(activation.spiky() and s(':activity:changed') not in list(activation.constraint.signals())
```

Comment:
```suggestion
                sum(1 if any(activation.spiky() and self[":activity"].changed_signal() not in list(activation.constraint.signals())
```

Reply:
Oh yes, this eliminates a possible future bug :+1: